staging/src/k8s.io/apiserver/pkg/server/config.go

```diff
@@ -376,6 +382,38 @@ type CompletedConfig struct {
 	*completedConfig
 }
 
+// AddPostStartHook allows you to add a PostStartHook that will later be added to the server itself in a New call.
+// Name conflicts will cause an error.
+func (c *Config) AddPostStartHook(name string, hook PostStartHookFunc) error {
+	if len(name) == 0 {
+		return fmt.Errorf("missing name")
+	}
+	if hook == nil {
+		return nil
+	}
+	if c.DisabledPostStartHooks.Has(name) {
+		return nil
```

Comment:
could we log that a hook is not going to be registered/started because it was explicitly disabled?